docs/Configuring_openshift_virtualization_vm_for_dpdk.md

```diff
@@ -0,0 +1,281 @@
+Doing so will result in the creation of a [network-attachment-definition](https://docs.openshift.com/container-platform/4.12/virt/virtual_machines/vm_networking/virt-attaching-vm-multiple-networks.html) that will be used by the OpenShift Virtualization VM in the next steps.
+- [Checking](https://docs.openshift.com/container-platform/4.12/post_installation_configuration/machine-configuration-tasks.html#checking-mco-status_post-install-machine-configuration-tasks) the node has a machineConfigPool, and [creating](https://access.redhat.com/solutions/5688941) one if needed.
+- Creating a [performance-profile](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/cnf-create-performance-profiles.html) using the [Node-Tuning-Operator](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/using-node-tuning-operator.html).
+Doing so will create a [runtimeClass](https://kubernetes.io/docs/concepts/containers/runtime-class/) that would be used by the OpenShift Virtualization VM in the next steps.
+
+In order to configure the cluster according to these steps please follow this OCP documentation.
+
+## OpenShift Virtualization VM configuration
+
+### Kubevirt cluster wide configuration
+
+The pod running the VM (a.k.a. the virt-launcher pod) needs to run the DPDK optimized performance profile defined in the previous chapter. This is done by selecting the runtimeclassName in the pod running the VM.
+Since currently Kubevirt does not support setting the runtimeclassName per VM, the change needs to be on a cluster level, by using the [jsonpatch annotations](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/docs/cluster-configuration.md#jsonpatch-annotations) on HCO:
+```bash
+oc annotate --overwrite -n openshift-cnv hco kubevirt-hyperconverged \
+  kubevirt.kubevirt.io/jsonpatch='[{"op": "add", \
+    "path": "/spec/configuration/DefaultRuntimeClassName", \
+    "value": <runtimeclass-name>}]'
+```
+
+### VM configuration
+
+When configuring the VM spec, the following changes need to be made.
+The changes are explained on each subchapter with appropriate snippets. In the end, an example VM template is offered.
+
+#### Requesting the SR-IOV interfaces
+
+If for example the created SR-IOV [network-attachment-definition](https://docs.openshift.com/container-platform/4.12/virt/virtual_machines/vm_networking/virt-attaching-vm-multiple-networks.html) name is `dpdk-net`, then the VM template interface request snippet should be like this:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+spec:
+  template:
+    spec:
+...
+  domain:
+    interfaces:
+      - sriov: {}
+        name: nic-east
+        pciAddress: 0000:07:00.0
+  networks:
+    - name: nic-east
+      multus:
+        networkName: dpdk-net
```

Comment:
The indentation here is incorrect.

Reply:
DONE

Reply:
The indentation is still incorrect.

Reply:
I redid it now, but I don't think there was a problem. Are you sure there is an indentation issue?

Reply:
`domain` should be a child of `spec`:
http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachineinstancespec

In this example it is his sibling becasue of incorrect indentation.